src/login/loginctl.c

```diff
@@ -137,124 +139,201 @@ static int list_table_print(Table *table, const char *type) {
-                                        &leader_pid,
-                                        &class,
-                                        &tty,
-                                        &idle,
-                                        &idle_timestamp_monotonic,
-                                        /* object = */ NULL);
+                r = sd_bus_message_enter_container(reply, 'r', signature[level]);
                 if (r < 0)
                         return bus_log_parse_error(r);
                 if (r == 0)
                         break;
 
-                r = table_add_many(table,
-                                   TABLE_STRING, session_id,
-                                   TABLE_UID, (uid_t) uid,
-                                   TABLE_STRING, user,
-                                   TABLE_STRING, empty_to_null(seat),
-                                   TABLE_PID, (pid_t) leader_pid,
-                                   TABLE_STRING, class,
-                                   TABLE_STRING, empty_to_null(tty),
-                                   TABLE_BOOLEAN, idle);
-                if (r < 0)
-                        return table_log_add_error(r);
-
-                if (idle)
-                        r = table_add_cell(table, NULL, TABLE_TIMESTAMP_RELATIVE_MONOTONIC, &idle_timestamp_monotonic);
-                else
-                        r = table_add_cell(table, NULL, TABLE_EMPTY, NULL);
+                SessionStatusInfo i = {};
+                r = sd_bus_message_read(
+                                reply,
+                                "suss",
+                                &i.id,
+                                &i.uid,
+                                &i.name,
+                                &i.seat);
                 if (r < 0)
-                        return table_log_add_error(r);
-        }
-
-        r = sd_bus_message_exit_container(reply);
-        if (r < 0)
-                return bus_log_parse_error(r);
-
-        return 0;
-}
-
-static int list_sessions_table_add_fallback(Table *table, sd_bus_message *reply, sd_bus *bus) {
-
-        static const struct bus_properties_map map[] = {
-                { "Leader",                 "u", NULL, offsetof(SessionStatusInfo, leader)                        },
-                { "Class",                  "s", NULL, offsetof(SessionStatusInfo, class)                         },
-                { "TTY",                    "s", NULL, offsetof(SessionStatusInfo, tty)                           },
-                { "IdleHint",               "b", NULL, offsetof(SessionStatusInfo, idle_hint)                     },
-                { "IdleSinceHintMonotonic", "t", NULL, offsetof(SessionStatusInfo, idle_hint_timestamp.monotonic) },
-                {},
-        };
+                        return bus_log_parse_error(r);
 
-        int r;
+                const struct bus_properties_map *map = NULL;
+                if (level > LIST_SESSIONS_TRADITIONAL) {
```

Comment:
Not a big matter, but I'd like to make the condition
```c
if (level >= LIST_SESSIONS_EX)
```
as the fields read below is in Ex.